### ISSUE TITLE
Implement Minimum Data Requirements for IX Object Creation

### ISSUE THREAD

**Issue by martinhannigan** [peeringdb/peeringdb#1618]
**Is your feature request related to a problem? Please describe.**

Currently, an IX object can be created in the database without any supporting data, such as:

    Assigned prefixes,
    Peers,
    Facilities.

**Who is affected by the problem?**

Internet network operators and users of PeeringDB

**What is the impact?**

The creation of an IX object without adhering to industry standards and community-accepted minimums results in the insertion of objects that do not accurately represent the defined concept of an IX. This is/should be a problem.

**Are there security concerns?**

Perhaps.

**Are there privacy concerns?**

No. There are accuracy and competitive concerns.

**Describe the solution you'd like**

Implement IX object creation minimum logic to require:

    At least one assigned IPv4 prefix, and
    At least three affiliated net objects.

**Do you think this feature will require a formal design?**

Perhaps.

**Describe alternatives you've considered**

Doing nothing, which would allow inaccurate data to be entered into the database, negatively impacting its reliability.

**Could this feature request need support from the Admin Committee?**

No.

**What is the proposed priority?**

Normal

**Provide a rationale for any/all of the above**

The minimum definition of an Internet Exchange (IX) is generally considered to be at least three ASNs connected to a switch fabric. Two ASNs are widely considered a Private Network Interconnect (PNI) which does not require the communities attention or implying such resources are required to interconnect. The proposed logic for IX object creation aims to align with this liberal definition and practices used by most RIRs for IPv4 needs assessment during IX prefix assignments. 

Examples of new entries which provide challenging value to the PeeringDB database:

![Screenshot 2024-05-23 at 7 12 43 PM](https://github.com/peeringdb/peeringdb/assets/21081154/a4b8cd5f-7a8e-4262-9329-2d7cffe635b5)
![Screenshot 2024-05-23 at 7 12 35 PM](https://github.com/peeringdb/peeringdb/assets/21081154/00f51e4f-58be-4405-ad29-90893da65e3a)

**Comment by martinhannigan** [comment]
+1

**Comment by arnoldnipper** [comment]
You can't create an `ix` object w/o a prefix. An `ix` operator can only add networks to its IXP if the IXP provides an IX-F JSON and a network allows the IXP update. Iow: networks decide whether they want to be listed at an IXP or not.

**Comment by arnoldnipper** [comment]
-1

**Comment by grizz** [comment]
+1 to prefixes and facilities, both have been agreed on already.

Agree that peers cannot be required.

**Comment by jackcarrozzo** [comment]
+1 to prefixes and facilities

**Comment by martinhannigan** [comment]
+1

we should look at no peers issue. Really excited to see progress with
automation!

**Comment by arnoldnipper** [comment]
> we should look at no peers issue.

That goes against the **fundamental** paradigm that PeeringDB is user-maintained. We don't want to force `net` to register at an `ix`. However, we encourage every `net` to keep its presences up to date.

**Comment by arnoldnipper** [comment]
> +1 to prefixes and facilities, both have been agreed on already.

IIrc, we didn't agree on facilities, yet. If, which issue? Prefix (either v6 or v4) already is implemented. Hence, nothing to do, IMO

**Comment by arnoldnipper** [comment]
ATM, you can remove the last `ixpfx` from your `ix`. If you do so, the `ix` should be removed as well.

**Comment by martinhannigan** [comment]
Agree, if the data is changed after the fact, a deletion should occur. In IX can't be taken commercially serious with v6 only which is also too easy to apply so a valid facility is a stop gap. However it would be much stronger if it were v4 and a facility. YMMV but still +1.

**Comment by martinhannigan** [comment]
On Tue, Jun 11, 2024 at 9:15 AM Arnold Nipper ***@***.***>
wrote:

> Just for my own curiosity… are these things the AC has to squash all the
> time, new IXs with no peers or v6 only? Or is this more rare in practice?
>
> A request for a new ix must have a prefix (v4 or v6) which belongs to the
> requestor. This is already checked during registration.
> Peers/members/participants can't be added by the ix. So far, it is also
> no requirement to associate the ix with a fac. IMO, an IX operator has a
> high self-interest to show all of its presences at facilities (to attract
> new participants).
>
> AC checks, whether the data make sense and are consistent.
>

AC checks should = automation. Principally, a V6 prefix is too easy to get
globally and ab(use) to create a peeringDB entry. Requiring a bona-fide
facility ups that game. Requiring an IPv4 prefix, something that no IX can
credibly operate without, ups it further.

But most importantly, allowing facilities to delete bogus affiliations is
probably the compromise.

With the last bit, we may be missing a step: automated policing of
disaffiliated entries into some non visible state.

HTH,

-M<

**Comment by jackcarrozzo** [comment]
PC call: there are v6-only isps

**Comment by jackcarrozzo** [comment]
PC: sounds reasonable to do with v4 OR v6, not AND
